samples/secure-vertex-ai-gemini-poem/README.md

```diff
@@ -0,0 +1,115 @@
+# Secure Vertex AI Gemini Poem Demo
+
+This demo showcases the implementation of a secure Vertex AI Gemini Poem Demo which is available only to users authenticated with Google.
+
+## The Demo
+
+Demo asks Vertex AI Gemini LLM to write a short 1 paragraph poem, using the access token acquired during the OIDC authorization code flow
+
+### Setup
+
+This demo requires users to authenticate with Google.
+
+All you need to do is to register an application with Google, follow steps listed in the [Quarkus Google](https://quarkus.io/guides/security-openid-connect-providers#google) section.
+
+Name your Google application as `Quarkus LangChain4j AI`, and make sure an allowed callback URL is set to `http://localhost:8080/login`.
+Google will generate a client id and secret, use them to set `quarkus.oidc.client-id` and `quarkus.oidc.credentials.secret` properties:
+
+```properties
+quarkus.oidc.provider=google
+quarkus.oidc.client-id=${GOOGLE_CLIENT_ID}
+quarkus.oidc.credentials.secret=${GOOGLE_CLIENT_SECRET}
+quarkus.oidc.authentication.extra-params.scope=https://www.googleapis.com/auth/generative-language.retriever,https://www.googleapis.com/auth/cloud-platform
+quarkus.oidc.authentication.redirect-path=/login
+
+# See https://cloud.google.com/vertex-ai/docs/geeral/locations
+vertex-ai-region=europe-west2
+
+quarkus.langchain4j.vertexai.gemini.location=${vertex-ai-region}
+quarkus.langchain4j.vertexai.gemini.project-id=${GOOGLE_PROJECT_ID}
+```
+
+You must enable Vertex AI API in your Google Cloud project.
+
+Set `GOOGLE_PROJECT_ID` to the id of your Google Cloud project where you have registered the `Quarkus LangChain4j AI` application.
```

Comment:
Hold on, isn't it the same thing? `Quarkus LangChain4j AI` is the name of Google Cloud project itself.
Your wording seems to imply that you register applications "under" the project, which I don't think is true (but my knowledge of Google Cloud is very sketchy).

Reply:
@jmartisk I'm not sure how else to say it. In Google Cloud, you create a project, and then inside that project you configure things like the application name, etc. For example, my test project name is actually named `Quarkus` - here I have OAuth2 Consent Screen which has a different Application name like `Quarkus LangChain4J AI`, etc, different APIs are enabled.

Shall we we just say `Set`GOOGLE_PROJECT_ID to the id of your Google Cloud project` ?

Reply:
Yeah I think "Set GOOGLE_PROJECT_ID to the id of your Google Cloud project" would be clearer :)

Reply:
Sure, let me simplify

Reply:
Done